### PR TITLE
Category banner

### PR DESCRIPTION
**Pull request by Sanjaybhattwebkul**
## Issue Reference
#7202
> Option added for upload category  banner

### CODE REVIEW

File: packages/Webkul/Admin/src/Resources/lang/en/app.php [hunk 1]
```diff
@@ -756,6 +756,8 @@
             'meta_description'         => 'Meta Description',
             'meta_keywords'            => 'Meta Keywords',
             'image'                    => 'Image',
+            'category_banner'          => 'Category Banner',
+            'banner-size'              => 'Banner resolution should be like 1920*350',
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/src/Resources/lang/en/app.php:760]: Use `_` and be consistent. You use one time `-` and one time `_`.

File: packages/Webkul/Admin/src/Resources/views/catalog/categories/create.blade.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
     {{ __('admin::app.catalog.categories.add-title') }}
 @stop
 
+
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/src/Resources/views/catalog/categories/create.blade.php:7]: Remove one line.

File: packages/Webkul/Category/src/Database/Migrations/2023_01_16_163456_add_category_banner_to_categories_table.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        if (!Schema::hasColumn('categories', 'category_banner'))
+        {
+            Schema::table('categories', function (Blueprint $table)
```
Comment (devansh-webkul) [inline packages/Webkul/Category/src/Database/Migrations/2023_01_16_163456_add_category_banner_to_categories_table.php:18]:
~~~php
Schema::table('categories', function (Blueprint $table) {
~~~

File: packages/Webkul/Category/src/Database/Migrations/2023_01_16_163456_add_category_banner_to_categories_table.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        if (Schema::hasColumn('categories', 'category_banner'))
+        {
+            Schema::table('categories', function (Blueprint $table)
```
Comment (devansh-webkul) [inline packages/Webkul/Category/src/Database/Migrations/2023_01_16_163456_add_category_banner_to_categories_table.php:34]: Same here.

File: packages/Webkul/Category/src/Models/Category.php [hunk 5]
```diff
@@ -66,7 +66,7 @@ class Category extends TranslatableModel implements CategoryContract
      *
      * @var array
      */
-    protected $appends = ['image_url', 'category_icon_url'];
+    protected $appends = ['image_url', 'banner_url','category_icon_url'];
```
Comment (devansh-webkul) [inline packages/Webkul/Category/src/Models/Category.php:69]: Space after comma.

File: packages/Webkul/Category/src/Repositories/CategoryRepository.php [hunk 6]
```diff
@@ -48,7 +48,8 @@ public function create(array $data)
         $category = $this->model->create($data);
 
         $this->uploadImages($data, $category);
-
+         $this->uploadImages($data, $category, 'category_banner');
```
Comment (devansh-webkul) [inline packages/Webkul/Category/src/Repositories/CategoryRepository.php:51]: Indentation.

File: packages/Webkul/Ui/src/Resources/assets/sass/app.scss [hunk 7]
```diff
@@ -1513,6 +1513,29 @@ modal {
     }
 }
 
+
+.large-image-wrapper {
+
```
Comment (devansh-webkul) [inline packages/Webkul/Ui/src/Resources/assets/sass/app.scss:1518]: Remove line.

File: packages/Webkul/Ui/src/Resources/assets/sass/app.scss [hunk 8]
```diff
@@ -1513,6 +1513,29 @@ modal {
...
+            height: 15%;
+            right: 0;
+            margin-right: 0;
+            text-align: center;
+            font-size: larger;
+            padding-top: 15px;
+
+        }
+    }
+
```
Comment (devansh-webkul) [inline packages/Webkul/Ui/src/Resources/assets/sass/app.scss:1536]: Remove line.

File: packages/Webkul/Velocity/src/Resources/views/shop/products/index.blade.php [hunk 9]
```diff
@@ -80,8 +80,8 @@
 
                     <div class="col-12 no-padding">
                         <div class="hero-image">
-                            @if (!is_null($category->image))
-                                <img class="logo" src="{{ $category->image_url }}" alt="" width="20" height="20" />
+                            @if (!is_null($category->category_banner))
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/index.blade.php:83]: Space after the exclamation mark.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

**Review by devansh-webkul — APPROVED** [review]

**Review by jitendra-webkul — APPROVED** [review]

**Pull request merged**